api/api_private_debug.go

```diff
@@ -45,39 +43,25 @@ func NewPrivateDebugAPI(b Backend) *PrivateDebugAPI {
 	return &PrivateDebugAPI{b: b}
 }
 
-// GetDBProperty returns the value of the given property of the given database.
-func (api *PrivateDebugAPI) GetDBProperty(dt database.DBEntryType, name string) string {
-	return api.b.ChainDB().GetProperty(dt, name)
-}
-
 // ChaindbProperty returns leveldb properties of the chain database.
 func (api *PrivateDebugAPI) ChaindbProperty(property string) (string, error) {
-	ldb, ok := api.b.ChainDB().(interface {
-		LDB() *leveldb.DB
-	})
-	if !ok {
-		return "", fmt.Errorf("chaindbProperty does not work for memory databases")
-	}
-	if property == "" {
-		property = "leveldb.stats"
-	} else if !strings.HasPrefix(property, "leveldb.") {
-		property = "leveldb." + property
-	}
-	return ldb.LDB().GetProperty(property)
+	return api.b.ChainDB().Stat(property)
 }
 
-// ChaindbCompact compacts the chain database if successful, otherwise it returns nil.
+// ChaindbCompact flattens the entire key-value database into a single level,
+// removing all unused slots and merging all keys.
 func (api *PrivateDebugAPI) ChaindbCompact() error {
-	ldb, ok := api.b.ChainDB().(interface {
-		LDB() *leveldb.DB
-	})
-	if !ok {
-		return fmt.Errorf("chaindbCompact does not work for memory databases")
-	}
-	for b := byte(0); b < 255; b++ {
-		logger.Info("Compacting chain database", "range", fmt.Sprintf("0x%0.2X-0x%0.2X", b, b+1))
-		err := ldb.LDB().CompactRange(util.Range{Start: []byte{b}, Limit: []byte{b + 1}})
-		if err != nil {
+	cstart := time.Now()
+	for b := 0; b <= 255; b++ {
```

Comment:
What does 'elapsed' mean? If `elapsed` means the elapsed time between the [start,end) database compaction, what about moving `cstart := time.Now()` inside the loop?

Reply:
Thank you! Updated